src/view/com/util/forms/PostDropdownBtn.tsx

```diff
@@ -344,6 +354,78 @@ let PostDropdownBtn = ({
     toggleReplyVisibility,
   ])
 
+  const onPressPin = useCallback(async () => {
+    const pinCurrentPost = !isPinned
+    let prevPinnedPost: string | undefined
+    try {
+      if (!currentAccount) throw new Error('Not logged in')
+      const {data: profile} = await agent.getProfile({
+        actor: currentAccount.did,
+      })
+      prevPinnedPost = profile.pinnedPost?.uri
+
+      updatePostShadow(queryClient, postUri, {pinned: pinCurrentPost})
+      if (prevPinnedPost && prevPinnedPost !== postUri) {
+        updatePostShadow(queryClient, prevPinnedPost, {pinned: false})
+      }
+
+      await profileUpdateMutate({
+        profile,
+        updates: existing => {
+          existing.pinnedPost = pinCurrentPost
+            ? {uri: postUri, cid: postCid}
+            : undefined
+          return existing
+        },
+      })
+
+      if (pinCurrentPost) {
+        Toast.show(_(msg`Post pinned`))
+      } else {
+        Toast.show(_(msg`Post unpinned`))
+      }
+
+      whenAppViewReady(agent, currentAccount.did, res => {
+        if (!res) return false
+        if (pinCurrentPost) {
+          return res.data.pinnedPost?.uri === postUri
+        } else {
+          return typeof res.data.pinnedPost === 'undefined'
+        }
+      }).then(() => {
```

Comment:
`profileUpdateMutate` does a `whenAppViewReady`, looks like you can pass a `checkCommitted` fn to do this check there. TIL :P

Reply:
woah, helpful!